library/core/src/intrinsics.rs

```diff
@@ -1071,6 +1071,7 @@ extern "rust-intrinsic" {
     // NOTE: While this makes the intrinsic const stable, we have some custom code in const fn
     // checks that prevent its use within `const fn`.
     #[rustc_const_stable(feature = "const_transmute", since = "1.46.0")]
+    #[cfg(not(bootstrap), rustc_diagnostic_item = "transmute")]
```

Comment:
Does this *have* to be `cfg(not(bootstrap))`? Seems like it would just be an unused diagnostic item otherwise which seems fine?

Reply:
It is an unused attribute otherwise, as we never even look at this without the change to the collector.

Reply:
diagnostic items are all used. Rustc doesn't know whether clippy is going to use them.